APE17.rst

```diff
@@ -0,0 +1,593 @@
+      sphinx-astropy
+    commands =
+      sphinx-build docs docs/_build/html -W -b html
+
+and this would be run using::
+
+    tox -e build_docs
+
+Developers not wishing to use tox could also accomplish the same by doing::
+
+    pip install -e .
+    cd docs
+    make html
+
+As for testing, we recognize that one of the benefits of the astropy-helpers infrastructure is
+that developers and users have come to expect that packages in the astropy
+ecosystem all implement ``python setup.py build_docs``, so packages that no
+longer rely on astropy-helpers could make the ``python setup.py build_docs``
+command print out information about how to run the tests, and
+optionally include a link to more information::
+
+    $ python setup.py build_docs
+
+    Note: building the documentation is no longer done using ``python setup.py build_docs``,
+    but instead you will need to run the following command:
+
+        tox -e build_docs
+
+    If you don't already have tox installed, you can install it with:
+
+        pip install tox
+
+    For more information, see https://docs.astropy.org/...
+
+    $
+
+We could then provide a clear documentation page in a centralized place on how
+to use tox to build the docs, or how to use the ``make html`` approach.
+
+These changes will have no impact on ReadTheDocs as that service never made use
+of the ``build_docs`` command, instead invoking sphinx directly with
+``sphinx-build``.
+
+Version helpers
+~~~~~~~~~~~~~~~
+
+Currently, astropy-helpers includes version helpers that take the version
+defined in ``setup.cfg`` or ``setup.py`` and add a developer string when in a
+checked out version of a git repository. The developer string consists of
+``version.devN`` where N is the number of commits since a release.
+
+The Python Packaging Authority (PyPA) now provide a setuptools extension called
+`setuptools_scm`_ which can entirely replace the version helpers in
+astropy-helpers. The way this package works is that versions are no longer
+specified in ``setup.cfg`` or ``setup.py`` - instead the versions are taken from
+tags. Developer version strings produced in this way are much more sophisticated
+and can indicate for example if the working copy is clean or has local changes,
+and whether it is a stable tagged version or a developer version. We note that
+the default way of setting up setuptools_scm results in the version string not
+being updated automatically in ‘editable’ installs of a package (i.e. ``pip
+install -e .``), however `a workaround
+<https://github.com/pypa/setuptools_scm/issues/273>`_ is available.
+
+Switching to `setuptools_scm`_ requires minimal configuration, which is well
+described in its documentation and we therefore do not repeat here. However, one
+subtlety is that since it relies on tags to determine versions, for packages
+such as the core astropy package which have all their tags on branches (at least
+in recent years), we will need to add a 'developer' tag on ``master`` straight
```

Comment:
@jdavies-st , does this sufficiently address your concern about dev tags?

Reply:
Yes, it does.  So that would be something like a `4.1.a0` tag on `master` so that the version `setuptools_scm` reports would be something meaningful, such as `astropy 4.1.a0.dev23` meaning that `master` is now 23 commits after version `4.0.0` was tagged (on its own branch).  That `a0` is important if you plan on using conda for anything with dev versions of astropy, but otherwise is probably not necessary.

Reply:
according to the version numbering PEP `.dev` is also fine for that, you would then get a version like `4.1.dev23`

Reply:
`astropy` never plans to have a dev release on `conda` though, so is the `a0` relevant then?

Reply:
No.  We were doing nightly dev builds of conda packages for `jwst`, so it was relevant to us.  Not relevant here though. This looks good to me. 👍